frontend/src/components/common/Resource/PortForward.tsx

```diff
@@ -0,0 +1,174 @@
+import { Box, Button, Tooltip } from '@material-ui/core';
+import MuiLink from '@material-ui/core/Link';
+import { Alert } from '@material-ui/lab';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import helpers from '../../../helpers';
+import { deletePortForward, listPortForward, startPortForward } from '../../../lib/k8s/apiProxy';
+import { getCluster } from '../../../lib/util';
+import Loader from '../Loader';
+
+interface PortForwardProps {
+  isPod: boolean;
+  containerPort: number;
+  namespace?: string;
+  name?: string;
+  isPodRunning?: boolean;
+}
+
+const PODS_BACKEND_PORTFORWARD_KEY = 'pods';
+const SERVICE_BACKEND_PORTFORWARD_KEY = 'services';
+const PORTFORWARD_STORAGE_KEY_FOR_PODS = 'portforward-pods';
+const PORTFORWARD_STORAGE_KEY_FOR_SERVICES = 'portforward-services';
+
+export interface PortForwardState {
+  id: string;
+  namespace: string;
+  cluster: string;
+  port: string;
+}
+
+export default function PortForward(props: PortForwardProps) {
+  const { isPod, containerPort, namespace, name, isPodRunning } = props;
+  const [isCurrentPodOrServicePortForwarding, setIsCurrentPodOrServicePortForwarding] =
+    React.useState(false);
+  const [error, setError] = React.useState(null);
+  const [portForward, setPortForward] = React.useState<PortForwardState | null>(null);
+  const cluster = getCluster();
+  const { t } = useTranslation();
+
+  if (!helpers.isElectron()) {
+    return null;
+  }
+  function handlePortForward() {
+    if (!namespace || !containerPort || !cluster) {
+      return;
+    }
+    setError(null);
```

Comment:
Nit: Please add a newline before and after this block.

Reply:
fixed